M365/src/DeleteContactsFromFolder/DeleteContactsFromFolder.ps1

```diff
@@ -0,0 +1,188 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.NOTES
+	Name: DeleteContactsFromFolders.ps1
+    Major Release History:
+        04/27/2023  - Initial Public Release on CSS-Exchange
+#>
+
+$BuildVersion = ""
+
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Host.ps1
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Verbose.ps1
+. $PSScriptRoot\..\..\..\Shared\OutputOverrides\Write-Warning.ps1
+. $PSScriptRoot\..\..\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+. $PSScriptRoot\..\..\..\Shared\LoggerFunctions.ps1
+. $PSScriptRoot\..\..\..\Shared\AzureFunctions\Get-GraphAccessToken.ps1
+. $PSScriptRoot\..\..\..\Shared\AzureFunctions\Invoke-GraphApiRequest.ps1
+
+function Write-DebugLog($Message) {
+    $Script:Logger = $Script:Logger | Write-LoggerInstance $Message
+}
+
+function Main {
+    if (Test-ScriptVersion -AutoUpdate -Confirm:$false) {
+        Write-Host ("Script was updated. Please re-run the command") -ForegroundColor Yellow
+        return
+    }
+
+    Write-Host ("DeleteContactsFromFolders.ps1 script version $($BuildVersion)") -ForegroundColor Green
+
+    $graphApiUrl = "https://graph.microsoft.com"
+    $unableToDeleteAllContacts = $false
+    $contacts = New-Object 'System.Collections.Generic.List[object]'
+
+    $getAccessTokenParams = @{
+        AzureADEndpoint = "https://login.live.com"
+        ClientId        = "7e791a92-18c9-4811-b48e-a0200052991c"
+        Scope           = "contacts.readwrite user.read mail.read"
+    }
+    $token = Get-GraphAccessToken @getAccessTokenParams
+
+    # Graph API call to get the current logged in user
+    $loggedInUserParams = @{
+        Query       = "me"
+        AccessToken = $token.AccessToken
+        GraphApiUrl = $graphApiUrl
+    }
+    $loggedInUserResponse = Invoke-GraphApiRequest @loggedInUserParams
+
+    if ($loggedInUserResponse.Successful -eq $false) {
+        Write-Host "Unable to query the logged in user. Please try again." -ForegroundColor Red
+        exit
+    }
+    Write-Host "Logged in user is: $($loggedInUserResponse.Content.UserPrincipalName)"
+
+    try {
+        # Get the list of contact folders
+        $listContactFolders = @{
+            Query       = "me/contactFolders"
+            AccessToken = $token.AccessToken
+            GraphApiUrl = $graphApiUrl
+        }
+        $listFoldersResponse = Invoke-GraphApiRequest @listContactFolders
```

Comment:
I see in `Invoke-GraphApiRequest` we don't provide the context of the possible error other than a status code. We should be able to pull out the content or something from the response if possible and have it displayed so it is made aware of what the issue is.

Reply:
Done. It should now be logged and can be used for troubleshooting purposes. I'm not sure if we really want to write out the content here as this script will be used by consumer users. Having it in the log should be sufficient IMO as those users will open a case with consumer support instead of troubleshooting the script. What do you think?